lib/raygun.batch.ts

```diff
@@ -1,31 +1,33 @@
 import { sendBatch } from "./raygun.transport";
 import { startTimer } from "./timer";
 import type { IncomingMessage } from "http";
-import {
-  callVariadicCallback,
-  Callback,
-  HTTPOptions,
-  SendOptions,
-} from "./types";
+import { HTTPOptions, SendOptions } from "./types";
 
 const debug = require("debug")("raygun");
 
-export type MessageAndCallback = {
+export type MessageAndPromise = {
   serializedMessage: string;
-  callback: Callback<IncomingMessage> | null;
+  promise: PromiseReference<IncomingMessage>;
 };
 
 export type PreparedBatch = {
   payload: string;
   messageCount: number;
-  callbacks: Array<Callback<IncomingMessage> | null>;
+  promises: Array<PromiseReference<IncomingMessage>>;
 };
 
 export type BatchState = {
-  messages: MessageAndCallback[];
+  messages: MessageAndPromise[];
   messageSizeInBytes: number;
 };
 
+// Contains the resolve and reject methods of the original promise
+// Allows to call to resolve and reject from outside the Promise scope
+type PromiseReference<T> = {
+  resolve: (message: T) => void;
+  reject: (error: Error) => void;
+};
+
```

Comment:
Should these constants be set to their default values in the raygun client initialization?

Reply:
This is not a client facing API